File: TripNest_Frontend/src/services/tripService.js
import API from "./api";

export const createTrip = (data) => API.post("/trips", data);

export const getMyTrips = (search, status, sort) => {
    const params = {};
    if (search) params.search = search;
    if (status) params.status = status;
    if (sort) params.sort = sort;
    return API.get("/trips", { params });
};

export const getTripById = (id) => API.get(`/trips/${id}`);

export const updateTrip = (id, data) => API.put(`/trips/${id}`, data);

export const deleteTrip = (id) => API.delete(`/trips/${id}`);

File: TripNest_Backend/src/main/java/com/tripnest/backend/service/TripService.java
package com.tripnest.backend.service;

import java.util.List;

import com.tripnest.backend.common.ApiResponse;
import com.tripnest.backend.dto.CreateTripRequest;
import com.tripnest.backend.dto.UpdateTripRequest;
import com.tripnest.backend.dto.response.TripResponse;

public interface TripService {

    ApiResponse<TripResponse> createTrip(CreateTripRequest request);

    ApiResponse<List<TripResponse>> getMyTrips(String search, String status, String sort);

    ApiResponse<com.tripnest.backend.dto.response.TripDetailsResponse> getTripById(Long id);

    ApiResponse<TripResponse> updateTrip(Long id, UpdateTripRequest request);

    ApiResponse<String> deleteTrip(Long id);

}